python/guide/module/gluon/activation.md

```diff
@@ -0,0 +1,241 @@
+
+## Activation Functions
+
+Deep neural networks are a way to express a nonlinear function with lots of parameters from input data to outputs. The nonlinearities that allow neural networks to capture complex patterns in data are referred to as activation functions. Over the course of the development of neural networks, several nonlinear activation functions have been introduced to make gradient-based deep learning tractable. 
+
+If you are looking to answer the question, 'which activation function should I use for my neural network model?', you should probably go with *ReLU*. Unless you're trying to implement something like a gating mechanism, like in LSTMs or GRU cells, then you should opt for sigmoid and/or tanh in those cells. However, if you have a working model architecture and you're trying to improve it's performance by swapping out activation functions or treating the activation function as a hyperparameter, then you may want to try hand-designed activations like SELU or a function discovered by reinforcement learning and exhaustive search like Swish. This guide describes these activation functions and others implemented in MXNet in detail.
+
+### Visualizing Activations
+In order to compare the various activation functions and to understand the nuances of their differences we have a snippet of code to plot the activation functions (used in the forward pass) and their gradients (used in the backward pass).
+
+
+```python
+import numpy as np
+import mxnet
+from mxnet import nd, gluon, autograd
+from matplotlib import pyplot as plt
+%matplotlib inline
+
+data = np.linspace(-10, 10, 501)
+
+def visualize_activation(data, activation_fn, activation):
```

Comment:
And in addition to removing data, can we infer activation name from activation_fn? Would only need single arg then.